### ISSUE TITLE
Documentation: how to manually position the tray with CSS?

### ISSUE THREAD

**Issue by asafratzon** [ealush/butter-toast#106]
Firstly thank you very much for this awesome package! a new v3 super-happy user.

It would be awesome to receive some usage examples of manual positioning of the tray (CSS).
or maybe support offsets from the ButterToasts' built-in positioning options?

Personally I need this when using a layout that include a header stripe which is fixed, and I would like the tray to be positioned at the top of the main container and not on the top of `<body>`.

Any assistance would be appreciated of how to accomplish the following:
Position the tray at the following:
Vertical: POSITION_TOP **- X pixels**
Horizontal: POSITION_CENTER

Thumbs up for the awesome work! much appreciated!

**Comment by ealush** [comment]
Hey @asafratzon, thanks for trying out this repo, and thanks for the feedback.

At the moment there's no built in way (other than css+`!importnant`) to set the position of the tray.

It seems to me like what you're actually looking for is a way to set the style properties of the tray. The POS_TOP/POS_CENTER... are just shortcuts I used to help the consumer specify what they want.

Internally, it does two things:
1. Set inline styles to the tray (top/right/left/bottom...)
2. Determine the stacking order (if it is a POS_BOTTOM tray the stacking order should be reversed so they push up instead of down).

Due to the second option, I cannot just allow `px` position, but I can do the following: Extend the inline styles I set on the tray with your inline styles, so when you do:

```js
<ButterToast
    position={{ vertical: POS_TOP, horizontal: POS_CENTER }}
    style={{ top: '20px' }}/>
```

I will extend the default styles with whatever it is you set in your style prop.

Does that seem like a reasonable solution to you?

**Comment by asafratzon** [comment]
Thanks for the quick response.

To make sure I understood it right, style prop is used to pass offsets that are applied in addition to the built-in positioning options?
So the '20 px' in your example will act as a shift of 20 px downward of the position of the tray?

If so it sounds good to me, hopefully general enough to be useful for others as well.

Many thanks.
Asaf.

**Comment by ealush** [comment]
Hey.
No, instead of adding it on top of my built-in style properties, I will extend it.

So internally, it would look somewhat like this:

```js
// Usage
<ButterToast
    position={{ vertical: POS_TOP, horizontal: POS_CENTER }}
    style={{ top: '20px' }}/>
```

```js
// Internal
Object.assign(tray.style, {
    position: 'absolute',
    top: '10px',
    left: '50%'
}, { top: '20px'});
```

So whatever you do that conflicts with my initial setting will override my own.
.

I'll have a version ready in the next couple of days :)

**Comment by asafratzon** [comment]
Thank you very much!

**Comment by ealush** [comment]
Hey @asafratzon, sorry for taking a bit longer than I initially planned. Had a busy week.

I just published `butter-toast@3.3.0` which should support your use case.
Let me know if that works for you.

### Using the `style` prop
If you deal with a more complex scenario in which you need some more customized styling over existing - such as displaying your notifications below your sticky header - you can add a `style` property to your tray and it will override Butter-Toast's styles with yours.

So for example:

```jsx
<ButterToast position={{vertical: POS_TOP,  horizontal: POS_CENTER}} style={{ top: '50px' }}>
```

Will override `POS_TOP`'s default `10px` with `50px`.

**Comment by asafratzon** [comment]
Works like a charm with 3.3.0!
Removed the custom css I had (`.butter-toast { top: 60px !important; }`) and used the style prop instead. The documentation is clear and hopefully will assist others.
Many thanks :-)
